### PR TITLE
Add suffix into image_name

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Review by zmiklank — CHANGES_REQUESTED** [review]
Hi, thanks for the PR.

However, I would suggest using two separate inputs for "suffix" and "tag".
This approach brings multiple features:
1) is generic - we aim to support also non-sclorg repos in the future
2) allows users to insert multiple tags
3) does not introduce any more conditionals and action steps
4) if it will be released in a version different than v1 we would not need to rewrite existing build-and-push actions in our containers.

Please think about this option.

**Comment by phracek** [comment]
Good, point, I have updated this pull request. It adds suffix.

**Comment by zmiklank** [comment]
@phracek, thanks for adjusting the PR.

I have one note, and I am not sure about the proceeding:
In step `check_exclude_file` we check if `.exclude-${{ inputs.tag }}` exists, however, that could be misleading, as logically, excluding mechanism has nothing to do with a tag as it is.

One way we can solve this is that we would check against  `.exclude-${{ inputs.suffix }}` and will let the user know in the description, that this input is used for 2 purposes.
Or we can leave it as it is, but then would be impossible to use more tags for the built image, and also it would be intransparent for the user.

WDYT?

**Comment by phracek** [comment]
> One way we can solve this is that we would check against `.exclude-${{ inputs.suffix }}` and will let the user know in the description, that this input is used for 2 purposes.
> Or we can leave it as it is, but then would be impossible to use more tags for the built image, and also it would be intransparent for the user.

What about if the suffix is empty?

**Comment by phracek** [comment]
A great approach would be to add a field like `docker_suffix` which will be used for checking Dockerfiles and .exclude files

**Comment by phracek** [comment]
This will pull request will add suffix for image_names and the other pull request will add a new field.

File: action.yml [hunk 1]
```diff
@@ -36,6 +36,10 @@ inputs:
     description: 'The action will use distgen for generating dockerfiles if true'
     required: false
     default: 'false'
+  suffix:
+    description: 'The action will add suffix to image name'
```
Comment (zmiklank) [inline action.yml]:
```suggestion
    description: 'A suffix to the image name. Also used for the excluding mechanism, see README.md'
```
I can adjust the README separately.

**Review by zmiklank — DISMISSED** [review]
Agreed, thanks!
I suggested a change in description, but otherwise LGTM.

File: action.yml [hunk 2]
```diff
@@ -36,6 +36,10 @@ inputs:
     description: 'The action will use distgen for generating dockerfiles if true'
     required: false
     default: 'false'
+  suffix:
+    description: 'Suffix name to image name'
```
Comment (zmiklank) [inline action.yml:40]:
```suggestion
    description: 'A suffix for the name of the built image'
```

File: action.yml [hunk 3]
```diff
@@ -75,12 +79,24 @@ runs:
...
 
-    - name: Check if .exclude-${{ inputs.tag }} is present in ${{ inputs.dockerfile_path }} directory
+    - name: Get suffix from ${{ inputs.dockerfile }}
+      id: df_suffix
+      shell: bash
+      run: |
+        if [ ${{ inputs.dockerfile }} == Dockerfile ]; then
+          suffix='centos7'
+        else
+          # See https://devhints.io/bash
```
Comment (zmiklank) [inline action.yml]:
```suggestion
```

I suggest removing this comment, but this suggestion can be ignored.

File: action.yml [hunk 4]
```diff
@@ -75,12 +79,24 @@ runs:
         pip3 -v install distgen
         DG=$HOME/.local/bin/dg make generate-all
 
-    - name: Check if .exclude-${{ inputs.tag }} is present in ${{ inputs.dockerfile_path }} directory
+    - name: Get suffix from ${{ inputs.dockerfile }}
+      id: df_suffix
+      shell: bash
+      run: |
+        if [ ${{ inputs.dockerfile }} == Dockerfile ]; then
+          suffix='centos7'
```
Comment (zmiklank) [inline action.yml:87]: Is there any possibility to rename the Dockerfiles for centos7 to `Dockerfile.centos7` in the future? This condition seems to be too sclorg-specific.

However, taking the suffix for the exclude mechanism from the Dockerfile name seems like the best solution to me and this condition is IMO unavoidable in the current situation.
Moreover, the whole exclude mechanism could be considered to be too sclorg-specific, so I would agree with bringing the centos7 condition to the `main` branch.

**Review by zmiklank — CHANGES_REQUESTED** [review]

File: action.yml (hunk 4 above)
Comment (phracek) [reply action.yml:87]: Let's discuss it on our meeting. It make sense to me at all? 
@hhorak @pkubatrh WDYT?

Comment (phracek) [reply action.yml:87]: I guess, for now, it can leave there and in the future, we can remove it as soon as `Dockerfile` disappears.

Comment (zmiklank) [reply action.yml:87]: OK, I agree.

And could you please squash the commits before merging?

Comment (phracek) [reply action.yml:87]: Sure

**Review by zmiklank — APPROVED** [review]

**Pull request merged**